src/Utilities/Microphysics/Microphysics.jl

```diff
@@ -0,0 +1,115 @@
+"""
+    Microphysics
+
+Microphysics parameterization functions (for example condensation,
+evaporation, autoconversion rates for cloud water and rain water)
+TODO: Snow, ice, and 2-moment formulation
+"""
+module Microphysics
+
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+
+using Printf
+
+# rates of conversion between microphysics categories
+export qv2qli, ql2qr, qr2qv
+
+# rain fall speed
+export terminal_velocity
+
+# conversion from specific humidity to mixing ratio (TODO - temporary)
+export q2r
+
+"""
+    qv2qli(PhasePartition, T, ρ[, timescale=1])
+
+Return the ql and qi source terms due to condesation/evaporation (for ql)
+and sublimation/resublimation (for qi)
+The source terms are obtained assuming a relaxation to equilibrum with a
+constant timescale. The timescale is an optional parameter.
+"""
+function qv2qli(q_sa::PhasePartition, q::PhasePartition,
+                timescale::DT=DT(1)) where {DT<:Real}
+
+    dqldt = (q_sa.liq - q.liq) / timescale
+    dqidt = (q_sa.ice - q.ice) / timescale
+
+  return (dqldt, dqidt)
+end
+
+"""
+    ql2qr(PhasePartition[, timescale=1, ql_0=1e-4])
+
+Return the qr source term due to collisions between cloud droplets
+(autoconversion) parametrized following Kessler (TODO - add reference).
+Contrary to the Kessler paper the process is not assumed to be instanteneous.
+The assumed timescale and autoconverion threshold ql_0 are optional parameters.
+"""
+function ql2qr(ql::DT, timescale::DT=DT(1), ql_0::DT=DT(5e-4)) where {DT<:Real}
+
+  dqrdt = max(DT(0), ql - ql_0) / timescale
+
+  return dqrdt
+end
+
+"""
+    q2r(q_, qt)
+
+Convert specific humidity to mixing ratio
+"""
+function q2r(q_::DT, qt::DT) where {DT<:Real}
+    return q_ / (DT(1) - qt)
+end
+
+"""
+    terminal_velocity(qt, qr, ρ, ρ_ground)
+
+Return rain terminal velocity.
+TODO - add citation
+"""
+function terminal_velocity(qt::DT, qr::DT, ρ::DT,
+                           ρ_ground::DT) where {DT<:Real}
+    rr  = q2r(qr, qt)
+    vel::DT = 0
+
+    if (qr > DT(1e-8)) # TODO - assert positive definite elswhere
+      vel = DT(14.34) * ρ_ground^DT(0.5) * ρ^-DT(0.3654) * rr^DT(0.1346)
```

Comment:
Please no constants anywhere except nondimensional numbers that actually are constant. Terminal velocity in the presence of buoyancy is ``sqrt(8 g r/C_d_hydrometeor (\rho_liquid/\rho_air - 1)``, where ``r`` is droplet radius and ``C_d_hydrometeor`` is the drag coefficient of the hydrometeor. ``C_d_hydrometeor`` should be a parameter in some parameter file. The radius can be, e.g., a separately defined effective radius. Probably the above relation put these pieces empirically together, but it thereby hides distributional assumptions made for the radius. We want to expose them, so we can carry them through consistently.

Reply:
OK - I'll rewrite this and test.